### PR TITLE
[VSEE-1244] [VSEE-1300] Adding trivy supply chain docs

### PR DESCRIPTION
**Pull request by rxl7906**
Summary:
* The following threads below will be resolved in the next docs PR. 
* Let's move forward with this docs change and merge it.

# Which other branches do you want a technical writer to cherry-pick this PR to (if any)?
N/A

It's best to PR to the most recent relevant branch and leave all the cherry-picking to the
writer, except where earlier branches require factual changes to the content.
For more information about the branches, see https://github.com/pivotal/docs-tap#branches

### CODE REVIEW

File: supply-chain/catalog/trivy-scanning.hbs.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Trivy Scanning Component
+
+## Description
+
+This document describes the Trivy Scanning Component for Tanzu Supply Chains. The Trivy Component provides out of the box trivy scanning that invokes [Supply Chain Security Tools - Scan 2.0](../../scst-scan/scan-2-0.hbs.md) in a Tanzu Supply Chain. See aquasecurity trivy [docs](https://github.com/aquasecurity/trivy) for more details on the trivy scanner.
```
Comment (rxl7906) [inline supply-chain/catalog/trivy-scanning.hbs.md:5]: TODO: should we note that this is v0.48.3 as specified in lever?

File: supply-chain/catalog/trivy-scanning.hbs.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+* To enable this Trivy Scanning Component in the Tanzu Supply Chain workload, the following packages must be installed:
+  * supply-chain.apps.tanzu.vmware.com
+  * supply-chain-catalog.apps.tanzu.vmware.com
+  * managed-resource-controller.apps.tanzu.vmware.com
+  * app-scanning.apps.tanzu.vmware.com
+  * source.component.apps.tanzu.vmware.com
+  * buildpack-build.component.apps.tanzu.vmware.com
+  * trivy.app-scanning.component.apps.tanzu.vmware.com
+* RBAC enabled for the default ServiceAccount for the DEV-NAMESPACE to the buildpacks
```
Comment (rxl7906) [inline supply-chain/catalog/trivy-scanning.hbs.md:58]: Should this be covered by the buildpack or supplychain docs?

File: supply-chain/catalog/trivy-scanning.hbs.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+* SERVICE-ACCOUNT-PUBLISHER is the service account that uploads results. It must have write access to scanResults.location.
+* spec.scanning.active-keychains is an array of enabled credential helpers to authenticate against registries using workload identity mechanisms.
+* WORKSPACE-BINDINGS are additional array of secrets, ConfigMaps, or EmptyDir volumes to mount to the running scan. The name is used as the mount path.
+* WORKSPACE-SIZE is size of the PersistentVolumeClaim the scan uses to download the image and vulnerability database.
+
+**Note:** See more examples of these IVS configurations [here](../../scst-scan/ivs-create-your-own.hbs.md).
+
+## Dependencies
+
+* To enable this Trivy Scanning Component in the Tanzu Supply Chain workload, the following packages must be installed:
```
Comment (rxl7906) [inline supply-chain/catalog/trivy-scanning.hbs.md:50]: The other teams named this using their general name but I felt this may be more accurate cause for example there's a supplychain v1 and supplychain v2 packages that get confusing due to the name.

File: supply-chain/catalog/trivy-scanning.hbs.md (hunk 1 above)
Comment (kkavitha) [reply supply-chain/catalog/trivy-scanning.hbs.md:5]: We should make it clear. Has it been set like this previously?

File: supply-chain/catalog/trivy-scanning.hbs.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+This document describes the Trivy Scanning Component for Tanzu Supply Chains. The Trivy Component provides out of the box trivy scanning that invokes [Supply Chain Security Tools - Scan 2.0](../../scst-scan/scan-2-0.hbs.md) in a Tanzu Supply Chain. See aquasecurity trivy [docs](https://github.com/aquasecurity/trivy) for more details on the trivy scanner.
+
+## API
+
+Component input: `image` is an input from the buildpack component
+
+Workload input: `spec.registry` is where the scan results are uploaded
+
+Workload input: `spec.source` is for the source component to retrieve source from git
```
Comment (kkavitha) [inline supply-chain/catalog/trivy-scanning.hbs.md:13]: Might be a knowledge gap from my part: Why do we need git source for trivy image scanning?

Comment (rxl7906) [reply supply-chain/catalog/trivy-scanning.hbs.md:13]: Thanks for raising this. I've been back and forth on this. I think something here needs to be revised.

The `spec.source` gets used in two places:
1) The input to our trivy supply chain component is `git` (shown [here](https://gitlab.eng.vmware.com/tanzu-application-platform/secure-tools/scanning/-/blob/main/packages/trivy.app-scanning.component.apps.tanzu.vmware.com/config/trivy-supply-chain-component.yaml#L529)) where `git` comes from `spec.source` that requires `git`. This `git` gets used to produce a `correlation-id` for the IVS. 
2) The `spec.source.git` gets used for the source component for git cloning. This probably belongs in the `Source Component` docs and not ours.

Technically the trivy supply chain component is using this `spec.source.git` for the `correlation-id` but I don't know if I should be mentioning a `correlation-id` here.

Long story short:
* My inclination now is to remove this line because it is confusing and I'll have to address this in a next docs PR which I'm working on now.

File: supply-chain/catalog/trivy-scanning.hbs.md (hunk 1 above)
Comment (rxl7906) [reply supply-chain/catalog/trivy-scanning.hbs.md:5]: According to our scanning [docs](https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Application-Platform/1.8/tap/scst-scan-integrate-app-scanning.html) for supplychain v1 we provided ClusterImageTemplates for Trivy/Grype but we didn't specify their versions.

I've been having some back and forth thoughts on this:
* The supply chains v1 and v2 provide a way to get started using what's available out of the box i.e.:
** supplychain v1 the ClusterImageTemplate Trivy/Grype are already provided and user can start trivy/grype scanning right away with the trivy/grype versions we provide
** supplchain v2 the Trivy Component is already provided and a user can start trivy scanning right away with the trivy version we provide
* However the supplychain v1 and v2 also allow the users to create/customize their own scanning experience:
** supplychain v1 users can create their own ClusterImageTemplate using whatever trivy/grype versions to scan with
** supplychain v2 users can create their own Component using whatever trivy version to scan with


Long story short:
* I feel like leaving this version specification out for now and letting this pass as a first draft. But I think I need to ask Ryan about this.

File: supply-chain/catalog/trivy-scanning.hbs.md (hunk 4 above)
Comment (kkavitha) [reply supply-chain/catalog/trivy-scanning.hbs.md]: ah makes sense. thanks!

**Review by kkavitha — APPROVED** [review]

**Comment by rxl7906** [comment]
Closing MR because docs are autogenerated here with this tool https://gitlab.eng.vmware.com/tanzu-application-platform/cartographer/experimental/component-docs. And here is what it looks like now https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Application-Platform/1.8/tap/supply-chain-reference-catalog-about.html
